stl/inc/span

```diff
@@ -0,0 +1,732 @@
+        return data() + size();
+    }
+
+    constexpr void swap(span& _Other) noexcept {
+        _Mydata.swap(_Other._Mydata);
+    }
+
+    // [span.objectrep] Views of object representation
+    _NODISCARD span<const byte, _Extent == dynamic_extent ? dynamic_extent : sizeof(element_type) * _Extent>
+        _As_bytes() const noexcept {
+        return {reinterpret_cast<const byte*>(data()), size_bytes()};
+    }
+
+    _NODISCARD span<byte, _Extent == dynamic_extent ? dynamic_extent : sizeof(element_type) * _Extent>
+        _As_writeable_bytes() const noexcept {
+        return {reinterpret_cast<byte*>(data()), size_bytes()};
+    }
+
+private:
+    _Span_storage<pointer, _Extent> _Mydata;
+};
+
+// [span.tuple] Tuple interface
+template <class _Ty, size_t _Extent>
+struct tuple_size<span<_Ty, _Extent>> : public integral_constant<size_t, _Extent> {};
+
+template <class _Ty>
+struct tuple_size<span<_Ty, dynamic_extent>>;
+
+template <size_t _Index, class _Ty, size_t _Extent>
+struct tuple_element<_Index, span<_Ty, _Extent>> {
+    static_assert(dynamic_extent != _Extent, "std::tuple_element<> not supported for std::span<T, dynamic_extent>");
+    static_assert(_Index < _Extent, "Index out of bounds in std::tuple_element<> (std::span)");
+    using type = _Ty;
+};
+
+template <size_t _Index, class _Ty, size_t _Extent>
+constexpr _Ty& get(span<_Ty, _Extent> _Span) noexcept {
+    static_assert(dynamic_extent != _Extent, "std::get<> not supported for std::span<T, dynamic_extent>");
+    static_assert(_Index < _Extent, "Index out of bounds in std::get<> (std::span)");
+    return _Span[_Index];
+}
+
+// FUNCTION TEMPLATE as_bytes
+template <class _Ty, size_t _Extent>
+auto as_bytes(span<_Ty, _Extent> sp) noexcept -> decltype(sp._As_bytes()) {
+    return sp._As_bytes();
+}
+
+// FUNCTION TEMPLATE as_writable_bytes
+template <class _Ty, size_t _Extent>
+auto as_writable_bytes(span<_Ty, _Extent> sp) noexcept
+    -> enable_if_t<!is_const_v<_Ty>, decltype(sp._As_writeable_bytes())> {
+    return sp._As_writeable_bytes();
+}
+
+// FUNCTION TEMPLATE swap
+template <class _Ty, size_t _Extent>
+constexpr void swap(span<_Ty, _Extent>& _Left, span<_Ty, _Extent>& _Right) noexcept {
+    _Left.swap(_Right);
+}
+
+// ALIAS TEMPLATES FOR DEDUCTION GUIDES
+template <class _Ty, size_t _Extent>
+span(_Ty (&)[_Extent])->span<_Ty, _Extent>;
+
+template <class _Ty, size_t _Extent>
+span(array<_Ty, _Extent>&)->span<_Ty, _Extent>;
+
+template <class _Ty, size_t _Extent>
+span(const array<_Ty, _Extent>&)->span<const _Ty, _Extent>;
+
+template <class _Container>
+span(_Container&)->span<typename _Container::value_type>;
+
+template <class _Container>
+span(const _Container&)->span<const typename _Container::value_type>;
+
+_STD_END
+
+#pragma pop_macro("new")
+_STL_RESTORE_CLANG_WARNINGS
+#pragma warning(pop)
+#pragma pack(pop)
+#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
+#pragma message("span is only available with C++20 or later.")
+#endif // !_HAS_CXX20

```

Comment:
Our endifs don't have the negation here.

Reply:
Ok